apps/verticalhome/js/app.js

```diff
@@ -158,6 +187,182 @@
+            pinnedItems.splice(x,1);
+          }
+        }
+
+        for(var z = 0; z < this.deletedApps.length; z++){
+          app.itemStore.deletePinnedAppItem(this.deletedApps[z]);
+        }
+      }
+
+      for(var i = 0; i<pinnedItems.length; i++){
+        var obj = {};
+
+        if(pinnedItems[i].entry!=null){
+
+          obj.name = pinnedItems[i].entry.name;
+          obj.manifestURL = typeof(
+            pinnedItems[i].entry.manifestURL
+            ) == 'undefined'?
+            pinnedItems[i].entry.targetApp.manifestURL
+            : pinnedItems[i].entry.manifestURL;
+
+          obj.icon = typeof(
+            pinnedItems[i].entry.icon
+            )=='undefined'?
+            pinnedItems[i].entry.icon.baseURI
+            :pinnedItems[i].entry.icon;
+
+          obj.index = pinnedItems[i].index;
+
+          app.itemStore.savePinnedAppItem(obj);
+
+        }
+      }
+
+       this.showNotif('changes-saved');
+
+    },
+
+    pinToMainScreen: function(){
+      var moreAppsSelected = '#more-apps-screen .selected';
+      var selMoreApp = document.querySelector(moreAppsSelected);
+      var selMoreAppIcon = selMoreApp.getAttribute('data-test-icon');
+      var selMoreAppDataId = selMoreApp.getAttribute('data-identifier');
+      var selMoreAppTitle = selMoreApp.querySelector('span.title').textContent;
+      var maxDataIndex = app.pinnedAppsManager.items.length;
+
+      app.pinnedAppsManager.addEntry( maxDataIndex,
+                                      selMoreAppDataId,
+                                      selMoreAppIcon,
+                                      selMoreAppTitle);
+
+      Array.prototype.slice.call(
+        document.querySelector('#more-apps-screen #icons').children
+      );
+
+      app.itemStore.applicationSource.
+        removeIconFromGrid( selMoreAppDataId );
+
+      this.renderGrid();
+
+      if (selMoreApp.parentNode) {
+        selMoreApp.parentNode.removeChild(selMoreApp);
+      }
+
+      this.hideMoreApps();
+
+      this.showNotif('added-to-home-screen');
+
+      var lastTimeSelected = app.pinnedAppsNavigation.currentIndex();
+      var insrertedIndex = app.pinnedAppsManager.items.length;
+      var scrollDiff = insrertedIndex - maxDataIndex;
+      var selectedPinnedApp = app.pinnedAppsNavigation
+                              .elements[lastTimeSelected];
+
+      selectedPinnedApp.classList.remove('selected');
+
+      for(var z = 0; z < scrollDiff; z++){
+        var itm = app.pinnedAppsNavigation.elements[0];
+        var cln_tmp = itm.cloneNode(true);
+        app.pinnedAppsNavigation.elemList.appendChild(cln_tmp);
+        itm.parentNode.removeChild(itm);
+      }
+
+      app.pinnedAppsNavigation.elements[1].classList.add('selected');
+
+      app.pinnedAppsNavigation.scrollNum();
+
+      app.pinnedAppsNavigation.refresh();
+
+    },
+
+    unpinFromMainScreen: function(){
+
+      var selPinApp = document.querySelector('#pinned-apps-list .selected');
```

Comment:
May be it is more useful to use id for selected item. As I know there is only one selected item on this page. And document.getElementById() works faster than document.querySelector().